rfc/0006-threat-model.md

```diff
@@ -0,0 +1,262 @@
+
+## Threat 3 - Attacker exploits misconfiguration of webhook to bypass
+An attacker, who has rights to create workloads in the cluster, is able to exploit
+a mis-configuration to bypass the intended security control
+
+**Mitigation**
+Regular reviews of webhook configuration catch issues.
+
+**To do**
+We may provide some mechanism to notify operators when a webhook is changed.
+Maybe the controller can watch its webhooks, fix when the configuration is
+changed and notify the operator.
+
+### Threat 4 - Attacker has rights to delete or modify the k8s webhook  object
+
+An attacker who has Kubernetes API access, has sufficient privileges to delete
+the webhook  object in the cluster.
+
+**Mitigation**
+RBAC rights are strictly controlled.
+
+**To do**
+I think most of the RBAC is not Kubewarden responsibility. But we can help our users if we:
+- Warning them in our docs and *suggest* some minimum RBAC to be used.
+- Provide a policy which detect RBAC changes and **maybe** block them.
+
+
+## Threat 5 - Attacker gets access to valid credentials for the webhook
+An attacker gains access to valid client credentials for the admission controller webhook
+
+**Mitigation**
+Webhook fails closed.
+
+Kubewarden is failed closed. Thus, we should be fine.
+
+## Threat  6 - Attacker gains access to a cluster admin credential
+
+An attacker gains access to a cluster-admin level credential in the kubernetes cluster.
+
+**Mitigation**
+N/A
+
+I cannot see how Kubewarden can help here.
+
+## Threat 7 - Attacker sniffs traffic on the container network
+An attacker who has access to the container network is able to sniff traffic
+between the API  server and the admission controller webhook.
+
+**Mitigation**
+Webhook uses TLS encryption for all traffic
+
+Kubewarden are safe. Because the webhook connections are encrypted.
+
+
+## Threat 8 - Attacker carries out a MITM attack on the webhook
+An attacker on the container network, who has access to the NET_RAW capability
+can  attempt to use MITM tooling to intercept traffic between the API server
+and admission  controller webhook.
+
+**Mitigation**
+Webhook mTLS authentication is used.
+
+**To do**
+Kubewarden should implement mutual TLS authentication
+We can add in the recommended policies from the `kubewarden-defaults` Helm
+chart a policy to drop the `NET_RAW` capability.
+
+### Threat 9 - Attacker steals traffic from the webhook via spoofing
+An attacker is able to redirect traffic from the API server which is intended
+for the admission  controller webhook by spoofing.
+
+**Mitigation**
+Webhook mTLS authentication is used.
+
+**To do**
+Kubewarden should implement mutual TLS authentication
+
+### Threat 10 - Abusing a mutation rule to create a privileged container
+An attacker is able to cause a mutating admission controller to modify a workload,
+such that  it allows for privileged container creation
+
+**Mitigation**
+All rules are reviewed and tested.
+
```

Comment:
[pod-privileged-policy](https://github.com/kubewarden/pod-privileged-policy) can also be used here

Reply:
As @raulcabello mentions, if an explicit pod-privileged-policy is installed (validating only), then this cannot happen. Regardless of whatever the mutating policy did, the validating policy has to approve the request.

This is a harder situation for example if a user deploys a policy that does mutation and validation in the same policy (e.g. to save time reducing requests made by the api server). If a policy does many things, then it's up to the policy author to make sure that this kind of situation does not happen.

Reply:
If our "PSP best practices" selection deploys the pod-privileged-policy, this is not going to be an issue. Even if the user provides a mutating policy that by design/mistake makes a container privileged.

Reply:
Yes, the "PSP best practices" includes the `pod-privileged-policy`.